### DISCUSSION TITLE
Passing a parameter once, and then not in further requests

### DISCUSSION THREAD

**Discussion by cpotey** [TanStack/query#5610] [General]
Hope this is the correct place for this! But have been using and loving Tanstack Query lately, it's made things a lot easier to work with!

I have a question: is there a way to have a query parameter on the first request of a useQuery, and then "remove it" (and cache an object returned in that result) for future requests?

For example, I want to make this useQuery call with the parameter for the first time only (I only want to do it once because this parameter adds onto load times, but provides more up-to-date data than without the parameter - which could then be used in future requests)
`http://api.example.com/roles/?sync_users=True`

Somewhere in the data returned will be a `users` object, which contains up-to-date information. I want to keep a hold of this over future requests, and for it to take precedent over other `users` objects that will be returned in future requests. 

So in further requests (because that call has been done once at the start), remove that parameter to speed things up again, to be like I was calling
`http://api.example.com/roles/`. This will return a `users` object, but not the one we want - which is the data wanted from the initial request

---

Is this where something like [initial query data](https://tanstack.com/query/v4/docs/react/guides/initial-query-data#initial-data-function) would come into play?

Would I need to somehow merge the `users` object data in the latest request, with the previous cached request results?

I hope this makes a smidge of sense to someone! 😅

**Comment by TkDodo** [comment]
1) pretty weird api 😅 
2) seems like two different queries to me, so: – `['users', { sync: true }]` vs. `['users', { sync: false }]` as the key.

The `sync: false` queryFn could then read from `queryClient.getQueryData(['users', { sync: true }])`, get the users from there and merge them with what they're getting from the actual api response, all within the queryFn.

**Comment by cpotey** [comment]
> 1. pretty weird api 😅
> 2. seems like two different queries to me, so: – `['users', { sync: true }]` vs. `['users', { sync: false }]` as the key.
> 
> The `sync: false` queryFn could then read from `queryClient.getQueryData(['users', { sync: true }])`, get the users from there and merge them with what they're getting from the actual api response, all within the queryFn.

1. Tell me about it 🫠
2. Thanks, that's a great shout - I haven't used `getQueryData` yet - will have a nose over that.
